tos/tos_jsonenums.go

```diff
@@ -0,0 +1,116 @@
+// generated by jsonenums -type=ToS; DO NOT EDIT
+
+package tos
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var (
+	_ToSNameToValue = map[string]ToS{
+		"CS3":         CS3,
+		"CS4":         CS4,
+		"CS5":         CS5,
+		"CS6":         CS6,
+		"CS7":         CS7,
+		"AF11":        AF11,
+		"AF12":        AF12,
+		"AF13":        AF13,
+		"AF21":        AF21,
+		"AF22":        AF22,
+		"AF23":        AF23,
+		"AF31":        AF31,
+		"AF32":        AF32,
+		"AF33":        AF33,
+		"AF41":        AF41,
+		"AF42":        AF42,
+		"AF43":        AF43,
+		"EF":          EF,
+		"Lowdelay":    Lowdelay,
+		"Throughput":  Throughput,
+		"Reliability": Reliability,
+		"Lowcost":     Lowcost,
+	}
+
+	_ToSValueToName = map[ToS]string{
+		CS3:         "CS3",
+		CS4:         "CS4",
+		CS5:         "CS5",
+		CS6:         "CS6",
+		CS7:         "CS7",
+		AF11:        "AF11",
+		AF12:        "AF12",
+		AF13:        "AF13",
+		AF21:        "AF21",
+		AF22:        "AF22",
+		AF23:        "AF23",
+		AF31:        "AF31",
+		AF32:        "AF32",
+		AF33:        "AF33",
+		AF41:        "AF41",
+		AF42:        "AF42",
+		AF43:        "AF43",
+		EF:          "EF",
+		Lowdelay:    "Lowdelay",
+		Throughput:  "Throughput",
+		Reliability: "Reliability",
+		Lowcost:     "Lowcost",
+	}
+)
+
+func init() {
+	var v ToS
+	if _, ok := interface{}(v).(fmt.Stringer); ok {
+		_ToSNameToValue = map[string]ToS{
```

Comment:
I assume we want to build up a map from string -> ToS here rather than hard coding it? If so, I think all you really want is:
```go
func init() {
  _tosNameToValue = make(map[string]ToS, len(_tosValueToName))
  for tos, tosString := range _tosValueToName {
    _tosNameToValue[tosString] = tos
  }
}
```